drivers/flash/flash_stm32_ospi.c

```diff
@@ -2168,9 +2168,17 @@ static int flash_stm32_ospi_init(const struct device *dev)
 
 	/* Send the instruction to read the SFDP  */
 	const uint8_t decl_nph = 2;
+/*
+ * When CONFIG_MISRA_SANE=y, this macro is forcing the size of raw[]
+ * with decl_nph = 2
+ * else ISO C90 forbids array 'raw' whose size cannot be evaluated
+ */
+#define JESD216_SFDP_SIZE_NPH (sizeof(struct jesd216_sfdp_header)	\
+				+ ((2) * sizeof(struct jesd216_param_header)))
```

Comment:
Can't we move definition in `drivers/flash/jesd216.h`, and replace faulty `JESD216_SFDP_SIZE()` ?

Reply:
I think that would be the right way to go.